### PR TITLE
MET Filter update for mAOD reprocessing

### PR DESCRIPTION
**Pull request by schoef**
- The HBHE Iso Noise filter flag is promoted to a separate boolean in mAOD. 
- The eeBadSC filter is added to the recommended MET filters flag.
- Thresholds for the ecalDeadCellFilter and the ecalBoundaryEnergyFilter are updated. The latter flag is added to mAOD
- Fixing the Flag_goodVertices in mAOD

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@schoef is all of this in 75X/76X as well?
If not, please submit a PR there.

**Comment by schoef** [comment]
@cmsbuild Please test

**Comment by schoef** [comment]
Slava: will do, one I figured in which sequence the new filter is missing.

**Comment by slava77** [comment]
@cmsbuild please test
(test requester list is not for all)

**Comment by slava77** [comment]
same issue with fastsim, apparently.

**Comment by schoef** [comment]
@slava77 
Should work now. Once tests are done I'll port to 75X,/76X.
This PR contains the MET filter recommendation for mAOD reprocessing

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/10406	 33e93df
- changes do not affect reco quantities.
    - Flag_HBHENoiseIsoFilter path is added
    - combined metFilters sequence has less filters now (hcalLaserEventFilter, goodVertices, trackingFailureFilter, ecalLaserCorrFilter, and trkPOGFilters are taken out from the combo).
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by slava77** [comment]
hold

we need equivalent changes in 75X/76X

**Comment by slava77** [comment]
Robert, please explain the changes with some brief note.
Probably, more specifically, if you are done with changing this PR.
Thanks.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@schoef please respond to 
https://github.com/cms-sw/cmssw/pull/10406#issuecomment-126909473
Thank you

**Comment by slava77** [comment]
The last updates added EcalDeadCellBoundaryEnergyFilter, which was never running in production before.
This should nominally go first through 76X IBs or even a pre-release relval cycle.
I'll run some more tests in 74X

**Comment by schoef** [comment]
@slava please hold, another patch is pending.

**Comment by schoef** [comment]
@slava77

**Comment by slava77** [comment]
@schoef 
are you done now with commits from your side?

**Comment by slava77** [comment]
... not starting the tests just yet due to a broken IB ...

**Comment by schoef** [comment]
@slava77 
Yes, I'm done. We had found that bug just this morning and I used this PR to sneak in the fix.

**Comment by slava77** [comment]
@schoef did you test it?
``` runTheMatrix.py -l 4.53 ``` would be enough in this case to get the problem

```
ValueError: Trying to override definition of primaryVertexFilter while it is used by the sequence metFilters
 new object defined in: CalibMuon/DTCalibration/ALCARECODtCalib_cff.py
 existing object defined in: RecoMET/METFilters/primaryVertexFilter_cfi.py
```
you can see in 
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/CalibMuon/DTCalibration/python/ALCARECODtCalib_cff.py

**Comment by slava77** [comment]
I was using 7_4_8 release

File: PhysicsTools/PatAlgos/python/slimming/metFilterPaths_cff.py [hunk 1]
```diff
@@ -2,16 +2,18 @@
...
+Flag_HBHENoiseIsoFilter = cms.Path(HBHENoiseFilterResultProducer * HBHENoiseIsoFilter)
 Flag_CSCTightHaloFilter = cms.Path(CSCTightHaloFilter)
 Flag_hcalLaserEventFilter = cms.Path(hcalLaserEventFilter)
 Flag_EcalDeadCellTriggerPrimitiveFilter = cms.Path(EcalDeadCellTriggerPrimitiveFilter)
-Flag_goodVertices = cms.Path(goodVertices)
+Flag_EcalDeadCellBoundaryEnergyFilter = cms.Path(EcalDeadCellBoundaryEnergyFilter)
+Flag_goodVertices = cms.Path(primaryVertexFilter)
 Flag_trackingFailureFilter = cms.Path(goodVertices + trackingFailureFilter)
 Flag_eeBadScFilter = cms.Path(eeBadScFilter)
 Flag_ecalLaserCorrFilter = cms.Path(ecalLaserCorrFilter)
```
Comment (slava77) [inline PhysicsTools/PatAlgos/python/slimming/metFilterPaths_cff.py]: there is supposed to be allMetFilterPaths with a corresponding modification in this file.
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/PhysicsTools/PatAlgos/python/slimming/metFilterPaths_cff.py#L29
the diff on the web is not showing it likely due to that change coming later than this one

**Comment by slava77** [comment]
-1
needs updates
in PV filter and allMetFilterPaths

**Comment by schoef** [comment]
OK, will do. It's nonsense that met filters are defined here
https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_X/CalibMuon/DTCalibration/python/ALCARECODtCalib_cff.py

I'll change that also.

**Comment by schoef** [comment]
OK, will make a new PR

**Pull request closed without merging**